throttler/throttler.go

```diff
@@ -0,0 +1,78 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+// ThrottledAction is a common interface for actions that can be executed
+type ThrottledAction interface {
+	Execute()
```

Comment:
It would be nice passing a `context` here in order to react to possible cancellations.